packages/expo-cli/src/commands/initAsync.ts

```diff
@@ -64,6 +64,10 @@ function assertValidName(folderName: string) {
       `Cannot create an app named ${chalk.red(`"${folderName}"`)}. ${validation}`
     );
   }
+  const isFolderNameForbidden = CreateApp.isFolderNameForbidden(folderName);
+  if (isFolderNameForbidden) {
+    throw new CommandError(`Cannot create an app named ${chalk.red(`"${folderName}"`)}.`);
```

Comment:
```suggestion
    throw new CommandError(`Cannot create an app named ${chalk.red(`"${folderName}"`)} because it would conflict with a dependency of the same name.`);
```